watcher/src/watcher_db.rs

```diff
@@ -209,6 +210,40 @@ impl WatcherDB {
             .collect::<Result<Vec<_>, WatcherDBError>>()?)
     }
 
+    /// Get the earliest timestamp for a given block.
+    /// The earliest timestamp reflects the time closest to when the block passed consensus.
+    /// If no timestamp is present, return u64::MAX, and a status code.
+    ///
+    /// Note: If there are no Signatures (and therefore no timestamps) for the given
+    ///       block, the result from get_signatures will be Ok(vec![]).
+    ///       A consensus validator only writes a signature for a block in which it
+    ///       participated in consensus. Therefore, if the watcher is only watching
+    ///       a subset of nodes, and those nodes happened to not participate in this
+    ///       block, the timestamp result will be unavailable for this block. It is
+    ///       also possible to be in a temporary state where there are no signatures
+    ///       for the given block if the watcher sync is behind the ledger sync.
+    pub fn get_block_timestamp(
+        &self,
+        block_index: u64,
+    ) -> Result<(u64, TimestampResultCode), WatcherDBError> {
```

Comment:
Is this better than `Result<Result<u64, TimestampResultCode>, WatcherDBError>`? That could remove the need for the `u64::MAX` thing and ensure the result code cannot be ignored.

Reply:
Oh interesting - I think that's a good suggestion, but my hesitation would be that I do like how explicit the behavior is currently, so that the caller doesn't have to transform the error.